compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2638,6 +2638,10 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     ref i => bug!("`impl Trait` pointed to non-opaque type?? {:#?}", i),
                 }
             }
+            hir::TyKind::ImplTraitInTrait(item_id) => {
+                let def_id = item_id.def_id.to_def_id();
+                tcx.mk_projection(def_id, InternalSubsts::identity_for_item(tcx, def_id))
```

Comment:
I find it very weird to be re-using projections here. I mean, we could (should?) merge projections and opaque types at some point, but for now this will require quite some work to get diagnostics back up to speed. What was the problem with using `ty::Opaque`?

Reply:
I guess the alternative is to fix all diagnostics now mentioning something other than the `impl Trait` in the trait. We can delay this to the future, but the current diagnostics are not great